test/lib/NonLinearDutchDecayLib.t.sol

```diff
@@ -474,4 +474,36 @@ contract NonlinearDutchDecayLibTest is Test, GasSnapshot {
         vm.expectRevert(NonlinearDutchDecayLib.InvalidDecayCurve.selector);
         mockNonlinearDutchDecayLibContract.decay(curve, startAmount, decayStartBlock, 1 ether, 1 ether);
     }
+
+    function testFuzzDutchDecayBeyondUint16Max(
+        uint256 startAmount,
+        uint256 blockDelta,
+        uint16 lastValidBlock,
+        uint256 decayAmountFuzz
```

Comment:
Consider making the fuzzed inputs match the inputs to decay(), and using vm.assume to trigger the overflow condition